webtau-core/src/main/java/com/twosigma/webtau/expectation/equality/handlers/ArrayAndIterableCompareToHandler.java

```diff
@@ -0,0 +1,92 @@
+package com.twosigma.webtau.expectation.equality.handlers;
+
+import com.twosigma.webtau.expectation.ActualPath;
+import com.twosigma.webtau.expectation.equality.CompareToComparator;
+import com.twosigma.webtau.expectation.equality.CompareToHandler;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class ArrayAndIterableCompareToHandler implements CompareToHandler {
+    @Override
+    public boolean handleEquality(Object actual, Object expected) {
+        return actual.getClass().isArray() && expected instanceof Iterable;
+    }
+
+    @Override
+    public void compareEqualOnly(CompareToComparator comparator, ActualPath actualPath, Object actual, Object expected) {
+        List<?> actualList = toList(actual);
+
+        comparator.compareUsingEqualOnly(actualPath, actualList, expected);
+    }
+
+    private List<?> toList(Object actual) {
```

Comment:
wow, this is pretty elaborate!

Reply:
It's only because, as far as I'm aware, you really can't do anything terribly generic with primitive arrays.

Reply:
@tsiq-karold what about Arrays.steam.box or something like that?

Reply:
I use that where I can but `Array.stream()` only has overrides for doubles, ints and longs.

Reply:
Ah, just noticed. Weird.

Reply:
Yes, it is: https://stackoverflow.com/questions/754294/convert-an-array-of-primitive-longs-into-a-list-of-longs#comment35809264_23117638

Even if there were other overrides, it would still require the switch statement and cast.  Would just mean I didn't have to write specific `toList` functions.